app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchTEIViewModel.kt

```diff
@@ -0,0 +1,268 @@
+package org.dhis2.usescases.searchTrackEntity
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import androidx.paging.PagedList
+import com.mapbox.geojson.FeatureCollection
+import java.util.HashMap
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.launch
+import org.dhis2.commons.data.SearchTeiModel
+import org.dhis2.commons.data.uids
+import org.dhis2.commons.network.NetworkUtils
+import org.dhis2.data.search.SearchParametersModel
+import org.dhis2.form.model.RowAction
+import org.dhis2.maps.geometry.mapper.featurecollection.MapCoordinateFieldToFeatureCollection
+import org.dhis2.maps.geometry.mapper.featurecollection.MapTeiEventsToFeatureCollection
+import org.dhis2.maps.geometry.mapper.featurecollection.MapTeisToFeatureCollection
+import org.dhis2.maps.mapper.EventToEventUiComponent
+import org.dhis2.maps.utils.DhisMapUtils
+import org.dhis2.usescases.searchTrackEntity.adapters.uids
+import org.hisp.dhis.android.core.program.Program
+
+class SearchTEIViewModel(
```

Comment:
I think that this class should be covered by Unit tests, at least asserting some expected `SearchTEScreenState` given different data

Reply:
I'm on it :)